### PR TITLE
feat!: add haskell

### PR DESCRIPTION
**Pull request by brandonspark**
I got bored and decided to add a language because I was blocked on some other things.

### Security

- [X] Change has no security implications (otherwise, ping the security team)

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -91,3 +91,9 @@
 [submodule "lang/semgrep-grammars/src/tree-sitter-clojure"]
 	path = lang/semgrep-grammars/src/tree-sitter-clojure
 	url = https://github.com/sogaiu/tree-sitter-clojure
+[submodule "lang/semgrep-grammars/src/semgrep-haskell"]
```
Comment (aryx) [inline .gitmodules:94]: this should not be here.

**Review by aryx — APPROVED** [review]
haha :)
Except for the extra submodule that is not needed, this looks fine.

**Comment by aryx** [comment]
test plan?
What are the current parsing stats?

**Comment by brandonspark** [comment]
@aryx i am not sure why CI is still failing, i thought it was the erroneous submodule

**Comment by brandonspark** [comment]
i also cannot test the parsing stats because im having difficulty using `make` in any context in my local set-up. not sure why. gonna have to fix that first.

**Comment by brandonspark** [comment]
It seems `ocaml-tree-sitter simplify` is causing grammar ambiguities in the resulting `grammar.json`, which makes me unable to test the parsing stats.

**Comment by mjambon** [comment]
> It seems `ocaml-tree-sitter simplify` is causing grammar ambiguities in the resulting `grammar.json`, which makes me unable to test the parsing stats.

See https://github.com/tree-sitter/tree-sitter/issues/1683

**Comment by mjambon** [comment]
@brandonspark you could try the workaround consisting in adding the unhidden rule that causes the ambiguity to the `inline` section (as described in tree-sitter/tree-sitter#1683).

**Comment by AllanSovereignJG** [comment]
It seems that this is a Git commit log of changes made to a repository that includes the addition of Haskell to the codebase. The initial change was made by @brandonspark, who requested a review from the team as a code owner, and later requested a review from specific individuals. @aryx approved the changes made in the commit and left a comment asking for the test plan and parsing stats. @mjambon later commented on a possible workaround for an issue related to grammar ambiguities.

The author of the commit, @brandonspark, made additional changes to the codebase in subsequent commits and eventually merged the changes into the main branch. One approving review was made by reviewers with write access to the repository.

**Pull request merged**